doc/examples_sphinx-gallery/stochastic_variability.py

```diff
@@ -0,0 +1,93 @@
+
+"""
+# %%
+# Import Libraries
+import igraph as ig
+import numpy as np
+import matplotlib.pyplot as plt
+import itertools
+
+# %%
+# First, we generate a graph.
+# Generates a random Erdos-Renyi graph (no clear community structure)
+def generate_random_graph(n, p):
+    return ig.Graph.Erdos_Renyi(n=n, p=p)
+  
+# %%
+# Generates a clustered graph with clear communities using the Stochastic Block Model (SBM)
+def generate_clustered_graph(n, clusters, intra_p, inter_p):
+    block_sizes = [n // clusters] * clusters
+    prob_matrix = [[intra_p if i == j else inter_p for j in range(clusters)] for i in range(clusters)]
+    return ig.Graph.SBM(sum(block_sizes), prob_matrix, block_sizes)
+
+# %%
+# Computes pairwise similarity (NMI, VI, RI) between partitions
+def compute_pairwise_similarity(partitions, method):
+    """Computes pairwise similarity measure between partitions."""
+    scores = []
+    for p1, p2 in itertools.combinations(partitions, 2):
+        scores.append(ig.compare_communities(p1, p2, method=method))
+    return scores
+
+# %%
+# Stochastic Community Detection
+# Runs Louvain's method iteratively to generate partitions
+# Computes similarity metrics: 
+def run_experiment(graph, iterations=50):
+    """Runs the stochastic method multiple times and collects community partitions."""
+    partitions = [graph.community_multilevel().membership for _ in range(iterations)]
+    nmi_scores = compute_pairwise_similarity(partitions, method="nmi")
+    vi_scores = compute_pairwise_similarity(partitions, method="vi")
+    ri_scores = compute_pairwise_similarity(partitions, method="rand")
+    return nmi_scores, vi_scores, ri_scores
+
+# %%
+# Parameters
+n_nodes = 100
+p_random = 0.05
+clusters = 4
+p_intra = 0.3  # High intra-cluster connection probability
+p_inter = 0.01  # Low inter-cluster connection probability
+
+# %%
+# Generate graphs
+random_graph = generate_random_graph(n_nodes, p_random)
+clustered_graph = generate_clustered_graph(n_nodes, clusters, p_intra, p_inter)
+
+# %%
+# Run experiments
+nmi_random, vi_random, ri_random = run_experiment(random_graph)
+nmi_clustered, vi_clustered, ri_clustered = run_experiment(clustered_graph)
+
+# %% 
+# Lets, plot the histograms
+fig, axes = plt.subplots(3, 2, figsize=(12, 10))
+measures = [(nmi_random, nmi_clustered, "NMI"), (vi_random, vi_clustered, "VI"), (ri_random, ri_clustered, "RI")]
+colors = ["red", "blue", "green"]
+
+for i, (random_scores, clustered_scores, measure) in enumerate(measures):
+    axes[i][0].hist(random_scores, bins=20, alpha=0.7, color=colors[i], edgecolor="black")
+    axes[i][0].set_title(f"Histogram of {measure} - Random Graph")
+    axes[i][0].set_xlabel(f"{measure} Score")
+    axes[i][0].set_ylabel("Frequency")
+    
+    axes[i][1].hist(clustered_scores, bins=20, alpha=0.7, color=colors[i], edgecolor="black")
+    axes[i][1].set_title(f"Histogram of {measure} - Clustered Graph")
+    axes[i][1].set_xlabel(f"{measure} Score")
```

Comment:
Could you please plot the probability density instead of counts? While doesn't make a difference here, it is generally good practice, and it becomes relevant when comparing datasets of different sizes.

Reply:
Also, please adjust the NMI and RI histograms to span the range $[0,1]$, and adjust the VI histogram to have a lower bound of 0.